src/components/sideMenu/view/sideMenuView.js

```diff
@@ -78,113 +71,109 @@ class SideMenuView extends Component {
     }
   };
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    const { isLoggedIn, accounts } = this.props;
-    const { isAddAccountIconActive } = this.state;
-
-    if (isLoggedIn !== nextProps.isLoggedIn) {
-      this._setMenuItems(nextProps.isLoggedIn);
+  useEffect(() => {
+    if (isAddAccountIconActive) {
+      setMenuItems(accounts);
     }
 
-    if (accounts !== nextProps.accounts && isAddAccountIconActive) {
-      this.setState({ menuItems: nextProps.accounts });
-    }
-  }
-
-  render() {
-    const { currentAccount, isLoggedIn, intl, handleLogout } = this.props;
-    const { menuItems, isAddAccountIconActive, storageT } = this.state;
-    const { buildVersion, appVersion } = VersionNumber;
-
-    return (
-      <View style={styles.container}>
-        <LinearGradient
-          start={{ x: 0, y: 0 }}
-          end={{ x: 1, y: 0 }}
-          colors={['#357ce6', '#2d5aa0']}
-          style={styles.headerView}
-        >
-          <ImageBackground source={SIDE_MENU_BACKGROUND} style={styles.imageBackground}>
-            {isLoggedIn && (
-              <View style={styles.headerContentWrapper}>
-                <UserAvatar username={currentAccount.name} size="xl" style={styles.userAvatar} />
-                <View style={styles.userInfoWrapper}>
-                  {currentAccount.display_name && (
-                    <Text numberOfLines={1} ellipsizeMode="tail" style={styles.username}>
-                      {currentAccount.display_name}
-                    </Text>
-                  )}
-                  <Text style={styles.usernick}>{`@${currentAccount.name}`}</Text>
-                </View>
-
-                <View style={styles.userInfoWrapper}>
-                  <IconButton
-                    name={isAddAccountIconActive ? 'ios-arrow-dropup' : 'ios-add-circle-outline'}
-                    color="white"
-                    size={20}
-                    onPress={this._handleOnPressAddAccountIcon}
-                    style={styles.addAccountIcon}
+    setMenuItems(isLoggedIn ? MENU.AUTH_MENU_ITEMS : MENU.NO_AUTH_MENU_ITEMS);
+
+    return () => {};
+  }, []);
```

Comment:
you should put isLoggedIn to this array. Because it was componentWillReceiveProps, it is mean this function was call every props or states changed. but default useEffect function just calling on componenDidMount. If you want to call every props and state changes, you should put this key to last array. So it should be [isLoggedIn]

Reply:
ok!